tests/integration/helm/install_test.go

```diff
@@ -0,0 +1,218 @@
+		ObjectMeta: kubeApiMeta.ObjectMeta{
+			Name: IstioNamespace,
+		},
+	}, kubeApiMeta.CreateOptions{}); err != nil {
+		_, err := cs.CoreV1().Namespaces().Get(context.TODO(), IstioNamespace, kubeApiMeta.GetOptions{})
+		if err == nil {
+			log.Info("istio namespace already exist")
+		} else {
+			t.Errorf("failed to create istio namespace: %v", err)
+		}
+	}
+
+	// Install base chart
+	err := h.InstallChart(BaseReleaseName, BaseChart,
+		IstioNamespace, overrideValuesFile)
+	if err != nil {
+		t.Errorf("failed to install istio %s chart", BaseChart)
+	}
+
+	// Install discovery chart
+	err = h.InstallChart(IstiodReleaseName, filepath.Join(ControlChartsDir, DiscoveryChart),
+		IstioNamespace, overrideValuesFile)
+	if err != nil {
+		t.Errorf("failed to install istio %s chart", DiscoveryChart)
+	}
+
+	// Install ingress gateway chart
+	err = h.InstallChart(IngressReleaseName, filepath.Join(GatewayChartsDir, IngressGatewayChart),
+		IstioNamespace, overrideValuesFile)
+	if err != nil {
+		t.Errorf("failed to install istio %s chart", IngressGatewayChart)
+	}
+
+	// Install egress gateway chart
+	err = h.InstallChart(EgressReleaseName, filepath.Join(GatewayChartsDir, EgressGatewayChart),
+		IstioNamespace, overrideValuesFile)
+	if err != nil {
+		t.Errorf("failed to install istio %s chart", EgressGatewayChart)
+	}
+}
+
+// deleteIstio deletes installed Istio Helm charts and resources
+func deleteIstio(t *testing.T, cs resource.Cluster, h *helm.Helm) {
+	scopes.Framework.Infof("cleaning up resources")
+	if err := h.DeleteChart(IngressReleaseName, IstioNamespace); err != nil {
+		t.Errorf("failed to delete %s release", IngressReleaseName)
+	}
+	if err := h.DeleteChart(EgressReleaseName, IstioNamespace); err != nil {
+		t.Errorf("failed to delete %s release", EgressReleaseName)
+	}
+	if err := h.DeleteChart(BaseReleaseName, IstioNamespace); err != nil {
+		t.Errorf("failed to delete %s release", BaseReleaseName)
+	}
+	if err := cs.CoreV1().Namespaces().Delete(context.TODO(), IstioNamespace, kubeApiMeta.DeleteOptions{}); err != nil {
+		t.Errorf("failed to delete %s namespace", IstioNamespace)
+	}
+}
+
+// verifyInstallation verify that the Helm installation is successfull
+func verifyInstallation(t *testing.T, ctx resource.Context, cs resource.Cluster) {
+	scopes.Framework.Infof("=== verifying istio installation === ")
+
+	retry.UntilSuccessOrFail(t, func() error {
+		if _, err := kubetest.CheckPodsAreReady(kubetest.NewSinglePodFetch(cs, IstioNamespace, "app=istiod")); err != nil {
+			return fmt.Errorf("istiod pod is not ready: %v", err)
+		}
+		if _, err := kubetest.CheckPodsAreReady(kubetest.NewSinglePodFetch(cs, IstioNamespace, "app=istio-ingressgateway")); err != nil {
+			return fmt.Errorf("istio ingress gateway pod is not ready: %v", err)
+		}
+		if _, err := kubetest.CheckPodsAreReady(kubetest.NewSinglePodFetch(cs, IstioNamespace, "app=istio-egressgateway")); err != nil {
+			return fmt.Errorf("istio egress gateway  pod is not ready: %v", err)
```

Comment:
nit: extra space between gateway and pod

Reply:
Addressed.